File: src/main/java/de/leximon/exsecratio/enchantments/DecayCurseEnchantment.java
package de.leximon.exsecratio.enchantments;

import net.minecraft.enchantment.Enchantment;
import net.minecraft.enchantment.EnchantmentTarget;
import net.minecraft.entity.EquipmentSlot;

public class DecayCurseEnchantment extends Enchantment {

    public DecayCurseEnchantment() {
        super(Enchantment.Rarity.VERY_RARE, EnchantmentTarget.BREAKABLE, EquipmentSlot.values());
    }

    @Override
    public int getMinPower(int level) {
        return 25;
    }

    @Override
    public int getMaxPower(int level) {
        return 50;
    }

    @Override
    public int getMaxLevel() {
        return 1;
    }

    @Override
    public boolean isTreasure() {
        return true;
    }

    @Override
    public boolean isCursed() {
        return true;
    }
}

File: src/main/kotlin/de/leximon/exsecratio/common/enchantments/DecayCurseEnchantment.kt
package de.leximon.exsecratio.common.enchantments

import net.minecraft.enchantment.Enchantment
import net.minecraft.enchantment.EnchantmentTarget
import net.minecraft.entity.EquipmentSlot

object DecayCurseEnchantment : Enchantment(Rarity.VERY_RARE, EnchantmentTarget.BREAKABLE, EquipmentSlot.values()) {

    override fun getMinPower(level: Int) = 25
    override fun getMaxPower(level: Int) = 50
    override fun getMaxLevel() = 1

    override fun isTreasure() = true
    override fun isCursed() = true
}
